From 79cd8d07986b579dbdf052449c3884a253b88f16 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@quansight.com>
Date: Tue, 7 Jul 2026 15:47:21 +0200
Subject: [PATCH 21/23] Set CPU flags via `add_definitions()`
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Set CPU flags via `add_definitions()` rather than adding them directly
to `CMAKE_CXX_FLAGS`.  This ensures that they are also passed to the
Objective C++ compiler (i.e. `CMAKE_OBJCXX_FLAGS`).

Signed-off-by: Michał Górny <mgorny@quansight.com>
---
 cmake/Codegen.cmake | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/cmake/Codegen.cmake b/cmake/Codegen.cmake
index e955e5bf679..2ce64496b0d 100644
--- a/cmake/Codegen.cmake
+++ b/cmake/Codegen.cmake
@@ -386,7 +386,7 @@ if(INTERN_BUILD_ATEN_OPS)
   list(APPEND CPU_CAPABILITY_FLAGS "${OPT_FLAG}")
 
   if(CXX_AVX512_FOUND)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_AVX512_CPU_DEFINITION")
+    add_compile_definitions("HAVE_AVX512_CPU_DEFINITION")
     # HIP code also needs CPU feature defines for DispatchStub ABI compatibility
     if(USE_ROCM)
       string(APPEND CMAKE_HIP_FLAGS " -DHAVE_AVX512_CPU_DEFINITION")
@@ -400,7 +400,7 @@ if(INTERN_BUILD_ATEN_OPS)
   endif(CXX_AVX512_FOUND)
 
   if(CXX_AVX2_FOUND)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_AVX2_CPU_DEFINITION")
+    add_compile_definitions("HAVE_AVX2_CPU_DEFINITION")
     if(USE_ROCM)
       string(APPEND CMAKE_HIP_FLAGS " -DHAVE_AVX2_CPU_DEFINITION")
     endif()
@@ -436,19 +436,19 @@ if(INTERN_BUILD_ATEN_OPS)
   endif(CXX_AVX2_FOUND)
 
   if(CXX_VSX_FOUND)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_VSX_CPU_DEFINITION")
+    add_compile_definitions("HAVE_VSX_CPU_DEFINITION")
     LIST(APPEND CPU_CAPABILITY_NAMES "VSX")
     LIST(APPEND CPU_CAPABILITY_FLAGS "${OPT_FLAG}  ${CXX_VSX_FLAGS}")
   endif(CXX_VSX_FOUND)
 
   if(CXX_ZVECTOR_FOUND)
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_ZVECTOR_CPU_DEFINITION")
+    add_compile_definitions("HAVE_ZVECTOR_CPU_DEFINITION")
     LIST(APPEND CPU_CAPABILITY_NAMES "ZVECTOR")
     LIST(APPEND CPU_CAPABILITY_FLAGS "${OPT_FLAG}  ${CXX_ZVECTOR_FLAGS}")
   endif(CXX_ZVECTOR_FOUND)
 
   if(CXX_SVE256_FOUND)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DHAVE_SVE_CPU_DEFINITION")
+    add_compile_definitions("HAVE_SVE_CPU_DEFINITION")
     list(APPEND CPU_CAPABILITY_NAMES "SVE256")
     list(APPEND CPU_CAPABILITY_FLAGS "${OPT_FLAG} -march=armv8-a+sve+bf16 -D__ARM_FEATURE_BF16 -msve-vector-bits=256")
     list(APPEND CPU_CAPABILITY_NAMES "SVE128")
